stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/messages/MessageListItemLiveData.kt

```diff
@@ -161,36 +175,52 @@ internal class MessageListItemLiveData(
         emptyList()
     }
 
+    /**
+     * Filters out or leaves in deleted messages based on their visibility
+     * set by the user.
+     */
+    private fun filterDeletedMessages(messages: List<Message>?): List<Message>? {
+        return when (deletedMessageVisibility.value) {
+            MessageListViewModel.DeletedMessageVisibility.ALL -> messages
+            MessageListViewModel.DeletedMessageVisibility.MINE -> messages?.filter { it.deletedAt == null || it.user.id == currentUser.value?.id }
+            MessageListViewModel.DeletedMessageVisibility.NONE -> messages?.filter { it.deletedAt == null }
+            else -> messages
```

Comment:
since both the else and the 1st branch are the same, how about just removing the first branch and leaving the last (else) clause?

Reply:
Removed the first branch.